BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -116,6 +116,47 @@ public T get(BlueKey key) throws BlueDbException {
 		return results;
 	}
 
+	// TODO make private?  or somehow enforce standard levels
+	public void rollup(long start, long end) throws BlueDbException {
+		List<File> filesToRollup = getFilesInRange(start, end);
+		Path path = Paths.get(segmentPath.toString(), start + "_" + end);
+		Path tmpPath = FileManager.createTempFilePath(path);
+		List<BlueEntity<T>> entities = new ArrayList<>();
+
+		// TODO switch to streaming
+		// TODO keep sorted, here and everywhere you write
+		for (File file: filesToRollup) {
+			List<BlueEntity<T>> es = fetch(file);
+			entities.addAll(es);

```

Comment:
I'd rather not have to hold all entities being rolled up in memory at one time. Instead, we could stream through each file one at a time while writing each object to the new temp file. Holding a write lock for awhile on that temp file shouldn't be an issue.

Reply:
using streaming in the latest updates